src/orion/plotting/backend_plotly.py

```diff
@@ -549,46 +549,139 @@ def build_frame():
+def build_parallel_frame(
+    experiments, names=None, with_evc_tree=True, order_by="objective"
+):
+    """Builds the dataframe for the parallel assessment"""
+    exp_parallels = dict(
+        name=list(), objective=list(), n_workers=list(), duration=list()
+    )
+    average_key = "objective,duration"
+    group = "n_workers"
+    x = group
+    x_title = "Number of workers"
+    y = "objective"
+    for i, experiment in enumerate(experiments):
+        dfs = experiment.to_pandas(with_evc_tree=with_evc_tree)
+        dfs = dfs.loc[dfs["status"] == "completed"]
+        if len(dfs) > experiment.max_trials:
+            dfs = dfs.head(experiment.max_trials)
+        if not dfs.empty:
+            start = dfs["suggested"].iloc[0]
+            end = dfs["completed"].iloc[-1]
+            duration = (end - start).seconds + 1
+
+            dfs = dfs.sort_values(order_by)
+            exp_parallels["duration"].append(duration)
+            name = names[i] if names else f"{experiment.name}-v{experiment.version}"
+            exp_parallels["name"].append(name)
+            exp_parallels[group].append(experiment.executor.n_workers)
+            exp_parallels["objective"].append(dfs["objective"].tolist()[0])
+
+    return pd.DataFrame(exp_parallels), group, average_key, x, y, x_title
+
+
+def build_durations_frame(
+    experiments, names=None, with_evc_tree=True, order_by="suggested"
+):
+    """Builds the dataframe for the duration plot"""
+    x = "duration"
+    x_title = "Experiment duration by second(s)"
+    frames, group, average_key, _, y, _ = build_regrets_frame(
+        experiments, names, with_evc_tree, order_by
+    )
+
+    return frames, group, average_key, x, y, x_title
+
+
+def build_regrets_frame(
+    experiments, names=None, with_evc_tree=True, order_by="suggested"
+):
+    """Builds the dataframe for regrets plot"""
+    frames = []
+    average_key = "best,duration"
+    group = "order"
+    x = group
+    x_title = f"Trials ordered by {order_by} time"
+    y = "best"
+    for i, experiment in enumerate(experiments):
+        df = experiment.to_pandas(with_evc_tree=with_evc_tree)
+        df = df.loc[df["status"] == "completed"]
+        df = df.sort_values(order_by)
+        if len(df) > experiment.max_trials:
+            df = df.head(experiment.max_trials)
+        df = orion.analysis.regret(df)
+
+        start = df["suggested"][0]
+        durations = [t.seconds + 1 for t in (df["completed"] - start)]
+        df["duration"] = durations
+        df["n_workers"] = experiment.executor.n_workers
+
+        df["name"] = [
+            names[i] if names else f"{experiment.name}-v{experiment.version}"
+        ] * len(df)
+        df[group] = list(range(len(df)))
+        frames.append(df)
+
+    return pd.concat(frames), group, average_key, x, y, x_title
+
+
+def regrets(
+    experiments,
+    with_evc_tree=True,
+    order_by="suggested",
+    build_frame_fn=build_regrets_frame,
```

Comment:
That's a great way of factoring it out! I think the regrets implementation will be better after this PR than it was before. Thanks! :)